### PR TITLE
[Restserver] Refine group api

### PR DESCRIPTION
**Pull request by ydye**
- Add leacy and deprecated comment for API to update singal value.
- Bind group api with /api/v2/groups
- Add api put /api/v2/groups to update a group's value.

### CODE REVIEW

File: src/rest-server/src/controllers/v2/group.js [hunk 1]
```diff
@@ -83,6 +83,54 @@ const createGroup = async (req, res, next) => {
...
+  try {
+    if (req.user.admin) {
+      let groupInfo = await groupModel.getGroup(groupname);
+      if (req.body.description !== '') {
+        groupInfo['description'] = req.body.description;
+      }
+      if (req.body.externalName !== '') {
+        groupInfo['externalName'] = req.body.externalName;
+      }
+      if (Object.keys(req.body.extension) > 0) {
```
Comment (mydmdm) [inline src/rest-server/src/controllers/v2/group.js:97]: So we could only set an `extension` item to `null` or empty, but could not delete the key, right?

Comment (mydmdm) [reply src/rest-server/src/controllers/v2/group.js:97]: actually here we are mixing the use of `put` (replacing, in other fields) and `patch` (modifying partially, in extensions). Refer to [here](https://stackoverflow.com/questions/24241893/should-i-use-patch-or-put-in-my-rest-api). We should unify the way we using `put` to update a map-like properties. Here are some options:
- use `put` to replace all dict
- use `put` like `patch` for dict objects, but clarify in documents
- add a parameter (e.g. `patchExtension=<true|false>`)

Comment (ydye) [reply src/rest-server/src/controllers/v2/group.js:97]: > So we could only set an `extension` item to `null` or empty, but could not delete the key, right?

We can left it as ```{}```

Comment (ydye) [reply src/rest-server/src/controllers/v2/group.js]: > * patchExtension
Has added a parameter ```patchExtension```. Its default value is ```true```

**Pull request merged**